### PR TITLE
sheldon 0.7.2

### PR DESCRIPTION
**Pull request by p-linnane**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by chenrui333** [comment]
```
  Broken dependencies:
    /opt/homebrew/opt/openssl@3/lib/libcrypto.3.dylib (openssl@3)
    /opt/homebrew/opt/openssl@3/lib/libssl.3.dylib (openssl@3)
```

**Comment by chenrui333** [comment]
Should we just allow the mixed openssl dependencies? cc @Homebrew/core

**Comment by ZhongRuoyu** [comment]
Can probably do this to make sure `openssl@1.1` is picked up:

https://github.com/Homebrew/homebrew-core/blob/ac63337a1bb3c370f1c6364f85c4e5cf41d8b244/Formula/cargo-edit.rb#L24-L27

By the way, it [vendors `libgit2`](https://github.com/rossmacarthur/sheldon/blob/0.7.2/Cargo.toml#L37). To use brewed `libgit2`, we need this:

https://github.com/Homebrew/homebrew-core/blob/ac63337a1bb3c370f1c6364f85c4e5cf41d8b244/Formula/cargo-edit.rb#L29-L38

**Comment by p-linnane** [comment]
Gave those changes a shot locally and returned this error:
```
Error: An exception occurred within a child process:
  NoMethodError: undefined method `sub' for nil:NilClass
```

File: Formula/sheldon.rb [hunk 1]
```diff
@@ -28,7 +28,21 @@ class Sheldon < Formula
...
+    # Ensure the declared `openssl@1.1` dependency will be picked up.
+    # https://docs.rs/openssl/latest/openssl/#manual
+    ENV["OPENSSL_DIR"] = Formula["openssl@1.1"].opt_prefix
+    ENV["OPENSSL_NO_VENDOR"] = "1"
+
+    # Read the default flags from `Cargo.toml` so we can remove the `vendored-libgit2` feature.
+    cargo_toml = (buildpath/"Cargo.toml").read
+    cargo_option_regex = /default\s*=\s*(\[.+?\])/m
+    cargo_options = JSON.parse(cargo_toml[cargo_option_regex, 1].sub(",\n]", "]"))
+    cargo_options.delete("vendored-libgit2")
```
Comment (branchvincent) [inline Formula/sheldon.rb:40]: this package is a little different: https://github.com/rossmacarthur/sheldon/blob/0.7.2/Cargo.toml#L37

an in-replace should work: 

```ruby
inreplace "Cargo.toml", /features = \["vendored-libgit2"\]/, "features = []"
```

**Comment by MikeMcQuaid** [comment]
> Should we just allow the mixed openssl dependencies?
No because they can cause segfaults

**Review by ZhongRuoyu — APPROVED** [review]

File: Formula/sheldon.rb [hunk 2]
```diff
@@ -28,6 +28,14 @@ class Sheldon < Formula
   end
 
   def install
+    # Ensure the declared `openssl@1.1` dependency will be picked up.
+    # https://docs.rs/openssl/latest/openssl/#manual
+    ENV["OPENSSL_DIR"] = Formula["openssl@1.1"].opt_prefix
+    ENV["OPENSSL_NO_VENDOR"] = "1"
+
+    # Replace vendored `libgit2` with our formula
+    inreplace "Cargo.toml", /features = \["vendored-libgit2"\]/, "features = []"
```
Comment (carlocab) [inline Formula/sheldon.rb:37]: You don't really need a regex here.
```suggestion
    inreplace "Cargo.toml", 'features = ["vendored-libgit2"]', "features = []"
```

**Pull request merged**